tests/test_core.py

```diff
@@ -1753,3 +1756,16 @@ def test_process_isconstant_2d():
     T_subseq_isconstant_comp = core.process_isconstant(T, m, T_subseq_isconstant)
 
     npt.assert_array_equal(T_subseq_isconstant_ref, T_subseq_isconstant_comp)
+
+
+def test_preprocess_diagonal_std_inverse():
+    T = np.random.rand(64)
+    m = 3
+    T[:m] = np.nan
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
```

Comment:
Is there a better way to catch warning for the sake of unit testing?

Reply:
What is it that you are trying to do?

Reply:
I am trying to add a test function to look for warnings and raise an error if it catches one that says "divide by zero encountered in divide"

Reply:
I think it might be more convenient to do this. So, here's an example test:

```
import pytest
import numpy as np

@pytest.mark.filterwarnings("error:divide by zero encountered in divide")
def test_something():
    a = np.ones(5, dtype=np.float64)
    b = np.zeros(5, dtype=np.float64)
    c = a / b
```

Notice that there are two parts:
1. `error` - turns whatever the ensuing warning is into an error 
2. the warning text to match for raising an error - notice that if you chaged `divide` to `multiply` then this does NOT raise an error and instead only warns us

Reply:
Thank you for the lesson!! Will revise the test function.